src/libstore/store-api.cc

```diff
@@ -834,7 +834,21 @@ std::map<StorePath, StorePath> copyPaths(ref<Store> srcStore, ref<Store> dstStor
                 MaintainCount<decltype(nrRunning)> mc(nrRunning);
                 showProgress();
                 try {
-                    copyStorePath(srcStore, dstStore, storePath, repair, checkSigs);
+                    if (dstStore->isTrusting || info->ca) {
+                        copyStorePath(srcStore, dstStore, storePath, repair, checkSigs);
+                    } else if (info->deriver && dstStore->storeDir == srcStore->storeDir) {
+                        auto drvPath = *info->deriver;
+                        auto outputMap = srcStore->queryDerivationOutputMap(drvPath);
+                        auto p = std::find_if(outputMap.begin(), outputMap.end(), [&](auto & i) {
+                            return i.second == storePath;
+                        });
+                        dstStore->buildPaths({{
+                            *info->deriver,
+                            p != outputMap.end() ? StringSet { p->first } : StringSet {},
+                        }});
+                    } else {
+                        dstStore->ensurePath(storePath);
```

Comment:
What happens if storePath can't be ensured by dstStore? Do we have some way to ask if it can build it, and if not, say they need to be trusted to build this path?

Reply:
This is the final fallback if we don't know how to build it. If this fails, we are out of options.

I suppose we could suggest trying without `trustless=true`, but since that's opt-in I wouldn't really worry.